src/Ng/CommandHelpers.cs

```diff
@@ -370,99 +399,78 @@ public static StorageFactory CreateSuffixedStorageFactory(string suffix, IDictio
             {
                 throw new ArgumentNullException("suffix");
             }
-
-            Uri storageBaseAddress = null;
-            string storageBaseAddressStr;
-            if (arguments.TryGetValue("-storageBaseAddress" + suffix, out storageBaseAddressStr))
+            try
             {
-                storageBaseAddressStr = storageBaseAddressStr.TrimEnd('/') + "/";
+                Uri storageBaseAddress = null;
+                string storageBaseAddressStr;
+                TryGetArgument(arguments, StorageBaseAddress + suffix, out storageBaseAddressStr);
+                if (!string.IsNullOrEmpty(storageBaseAddressStr))
+                {
+                    storageBaseAddressStr = storageBaseAddressStr.TrimEnd('/') + "/";
 
-                storageBaseAddress = new Uri(storageBaseAddressStr);
-            }
+                    storageBaseAddress = new Uri(storageBaseAddressStr);
+                }
 
-            string storageType;
-            if (!arguments.TryGetValue("-storageType" + suffix, out storageType))
-            {
-                TraceRequiredArgument("-storageType" + suffix);
-                return null;
-            }
+                string storageType;
+                TryGetArgument(arguments, StorageType, out storageType, required: true);
 
-            if (storageType.Equals("File", StringComparison.InvariantCultureIgnoreCase))
-            {
-                string storagePath;
-                if (!arguments.TryGetValue("-storagePath" + suffix, out storagePath))
+                if (storageType.Equals("File", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
This `if`, and the `Azure` `if` below, appear to be repeated in `CreateStorageFactory` as well. You may want to put it in a function.